iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtensionLogger.java

```diff
@@ -2,60 +2,148 @@
-    logMessage.put("mParticleEventId", requestId);
     String messageJson = gson.toJson(logMessage);
     System.out.println(messageJson);
   }
 
-  static void logApiTimeout(String url, UUID mparticleEventId) {
-    String eventIdString = mparticleEventId != null ? mparticleEventId.toString() : "Unavailable";
+  public void logIterableApiTimeout(String url, UUID mparticleEventId) {
+    String eventIdString = mparticleEventId != null ? mparticleEventId.toString() : "Error";
     Map<String, String> logMessage = new HashMap<>();
-    logMessage.put("errorType", "Retriable");
-    logMessage.put("message", "A timeout occurred while making request to Iterable");
+    logMessage.put("errorType", RETRIABLE_HTTP_ERROR);
+    logMessage.put("awsRequestId", awsRequestId);
+    logMessage.put("mparticleEventId", eventIdString);
+    logMessage.put("message", "Encountered a timeout while connecting to the Iterable API");
     logMessage.put("url", url);
+    String messageJson = gson.toJson(logMessage);
+    System.out.println(messageJson);
+  }
+
+  public void logProcessingError(String message, UUID mparticleEventId) {
+    Map<String, String> blobbyLogMessage = new HashMap<>();
+    blobbyLogMessage.put("mParticleBatch", mparticleBatch);
+    String blobbyId = logToBlobby(blobbyLogMessage);
+
+    String eventIdString = mparticleEventId != null ? mparticleEventId.toString() : "Error";
+    Map<String, String> logMessage = new HashMap<>();
+    logMessage.put("errorType", PROCESSING_ERROR);
+    logMessage.put("awsRequestId", awsRequestId);
     logMessage.put("mparticleEventId", eventIdString);
+    logMessage.put("blobbyId", blobbyId);
+    logMessage.put("message", message);
+    String messageJson = gson.toJson(logMessage);
+    System.out.println(messageJson);
+  }
+
+  public void logUnexpectedError(Exception e) {
+    Map<String, String> blobbyLogMessage = new HashMap<>();
+    blobbyLogMessage.put("mParticleBatch", mparticleBatch);
+    String blobbyId = logToBlobby(blobbyLogMessage);
+
+    Map<String, String> logMessage = new HashMap<>();
+    logMessage.put("errorType", UNEXPECTED_ERROR);
+    logMessage.put("awsRequestId", awsRequestId);
+    logMessage.put("blobbyId", blobbyId);
+    StringWriter sw = new StringWriter();
+    e.printStackTrace(new PrintWriter(sw));
+    logMessage.put("message", sw.toString());
     String messageJson = gson.toJson(logMessage);
     System.out.println(messageJson);
   }
 
-  public static void logError(String message) {
+  public void logMessage(String message) {
     Map<String, String> logMessage = new HashMap<>();
     logMessage.put("message", message);
+    logMessage.put("awsRequestId", awsRequestId);
     String messageJson = gson.toJson(logMessage);
     System.out.println(messageJson);
   }
+
+  public String getAwsRequestId() {
+    return awsRequestId;
+  }
+
+  public void setAwsRequestId(String id) {
+    awsRequestId = id;
+  }
+
+  public String getMparticleBatch() {
```

Comment:
Looks like this is not used anywhere, as well as `get/setAwsRequestId`. Do you expect this to be called at some point in the future? I'd remove these if they're not needed externally.

Reply:
Good call - I don't think they'll be needed, and we can add them back if so.